decidim-core/spec/system/editor_spec.rb

```diff
@@ -0,0 +1,1768 @@
+
+            drag("[data-image-resizer-control='top-right']", mode:, direction: "left", amount: 100)
+            expect(page).to have_selector("[data-image-resizer-dimension-value='#{width - 100}']", visible: :all)
+            expect(page).to have_selector("[data-image-resizer-dimension-value='#{height - 67}']", visible: :all)
+          end
+        end
+
+        context "with left side controls" do
+          it "allows resizing the image" do
+            drag("[data-image-resizer-control='bottom-left']", mode:, direction: "right", amount: 100)
+            expect_value(%(<div class="editor-content-image" data-image=""><img src="#{image_src}" alt="Test" width="#{dimensions[0] - 100}"></div>))
+
+            drag("[data-image-resizer-control='top-left']", mode:, direction: "left", amount: 50)
+            expect_value(%(<div class="editor-content-image" data-image=""><img src="#{image_src}" alt="Test" width="#{dimensions[0] - 50}"></div>))
+          end
+
+          it "removes the width attribute when resizing back to original width or above it" do
+            drag("[data-image-resizer-control='bottom-left']", mode:, direction: "right", amount: 100)
+            expect_value(%(<div class="editor-content-image" data-image=""><img src="#{image_src}" alt="Test" width="#{dimensions[0] - 100}"></div>))
+
+            drag("[data-image-resizer-control='top-left']", mode:, direction: "left", amount: 100)
+            expect_value(%(<div class="editor-content-image" data-image=""><img src="#{image_src}" alt="Test"></div>))
+
+            drag("[data-image-resizer-control='bottom-left']", mode:, direction: "right", amount: 100)
+            expect_value(%(<div class="editor-content-image" data-image=""><img src="#{image_src}" alt="Test" width="#{dimensions[0] - 100}"></div>))
+
+            drag("[data-image-resizer-control='top-left']", mode:, direction: "left", amount: 500)
+            expect_value(%(<div class="editor-content-image" data-image=""><img src="#{image_src}" alt="Test"></div>))
+          end
+        end
+      end
+
+      context "when mouse" do
+        it_behaves_like "resize controls", "mouse"
+      end
+
+      context "when touch" do
+        it_behaves_like "resize controls", "touch"
+      end
+
+      context "when the resize controls receive a click event" do
+        before do
+          # Focuses the image within the editor
+          prosemirror.native.send_keys [:left]
+        end
+
+        it "does not submit the form when resizing the image" do
+          page.find("[data-image-resizer-control='top-left']").click
+          page.find("[data-image-resizer-control='top-right']").click
+          page.find("[data-image-resizer-control='bottom-right']").click
+          page.find("[data-image-resizer-control='bottom-left']").click
+
+          expect(page).to have_current_path("/test_editor")
+        end
+      end
+    end
+
+    def drag(selector, mode: "mouse", direction: nil, amount: 0)
```

Comment:
Nice implementation. Maybe it would better to extract this, or a derivate function as stand alone ? It is the kind of method that would allow us to write specs for hompage content block ( reorder ) functionality.